### DISCUSSION TITLE
Welcome to Cbc Discussions!

### DISCUSSION THREAD

**Discussion by tkralphs** [coin-or/Cbc#331] [Discussion]
## 👋 Welcome!
  As you may have heard, Github has been developing a new feature called Discussions that is being beta-tested
  on select repositories. We're happy to announce that Cbc has been invited to be one of those repositories! We 
  envision Discussions as a way of eventually moving the kinds of conversations that currently take place on the 
  old Mailman list serve directly to where development is taking place, on Github! To take this thing for a test drive,
  please post here when you would otherwise post to the list serve. This a place to connect with other members of 
  our community on topics not suited for opening an Issue. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, feel free to comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by amas008** [comment]
Ted: Well done getting CBC onto the GitHub beta trial. This will be a good forum for [OpenSolver](http://opensolver.org) Excel users who have more technical questions about the CBC solver; I will let our users know. Also, it would be great to have a sticky post with instructions for Windows users on how to download the latest executable. Thanks for all your and Stefan's work. Andrew

**Reply by tkralphs** [reply]
I'm not too sure if we can make discussions sticky, but I'm now part of a group that can give feedback on Discussions directly to Github so I can certainly see if there are any plans for that.

**Comment by mjsaltzman** [comment]
Great to hear that this is finally moving forward. Any idea what the timeline is for pushing it out?

I'm wondering whether it makes sense to wait on this for the project mailing lists, rather than moving them to Google groups. The coin-announce and coin-discuss lists (and the board and council lists and probably some others) have no counterparts here, so I guess they need to be groups, but individual project lists might end up being moved twice in short order. What do you all think?

**Comment by tkralphs** [comment]
I'm not sure. This is a slow-moving train, but I can now see that there is indeed a roadmap and development activity around this feature. It's hard to predict when it roll out, could be a few months or could be much longer. If we can archive the old mailing lists over on Google Groups, it seems worthwhile to do that even if we don't use them for very long, since we don't want to commit to maintaining the Mailman archives in perpetuity and the archives should go somewhere. Seems like there is some work to do either way. I guess I would continue with it.

**Reply by svigerske** [reply]
The question may be which train is moving slower :).
The aim for the GitHub discussions seems to be to have them available by the end of this year: https://github.com/github/roadmap/issues/104

The mailman archives are just static HTML pages. Isn't it easy to keep them available even if the lists have been disabled? If the www.coin-or.org server isn't an option for that, then maybe some github pages repo.

**Reply by tkralphs** [reply]
Ideally, we would move the archives to here, but that may be too much to ask. I did see an open question in the support forums about he Discussions feature about migrating Google Groups archives. We'll see if there is an answer to that.

**Reply by tkralphs** [reply]
Hmm, just realized that I started a new thread instead of replying to Matt. Anyway, I missed your comment about Github's public roadmap. If that is accurate, I guess we don't have too long to wait, but we still need a good solution for archiving the mailing list. If it's just static HTML, then I guess we could dump them in gh-pages, as long as the archive is not too big (I guess it isn't).

**Comment by akazachk** [comment]
Great initiative, @tkralphs!

Having just heard of Discussions for the first time, I am wondering, is there a guideline for the type of questions that go under "Discussions" vs putting them under "Issues"? For example, maybe feature requests and the ensuing discussion more naturally go here.

By the way, my intro: my name is Aleksandr Kazachkov, and I am an active Clp and Cbc user, particularly in my work on developing new (disjunctive) cutting plane techniques. I have a Cgl class that is under development (for a few years now... 😅) that is available, but not very user-friendly or well-documented yet, here: https://github.com/akazachk/vpc. I am currently a postdoc in Montréal under Andrea Lodi, and I will be starting as an assistant professor at the University of Florida Department of Industrial and Systems Engineering in January 2021. I am interested in building an improved cut management layer for Cbc, among other things.

**Reply by tkralphs** [reply]
Eventually, there will be topic categories here like there are for issues (that feature is coming, but not available yet) and that will make it clearer what goes where. I would say things that require pretty concrete action (reproducible bugs, etc.) go in Issues, open-ended things that might inspire general discussion (requests for general information, questions, feature requests, etc.) would go here. But we also have the ability to transfer things back and forth so no matter where it goes at first, we will shepherd it to the right place. There will undoubtedly be situations where it's not very clear where something should go.

**Reply by akazachk** [reply]
Make sense! It is good they made it easy to transfer posts and cross-reference by having the numbering be consistent with the issues.

**Comment by h-g-s** [comment]
Hi @akazachk ,

Good to know that you are still planning to integrate your cutting planes 
into Cbc (Cgl more specifically). This work seems very promising !

I would like to suggest you to not wait too much to try to integrate it 
into Cbc, for the following reasons:

a) making it available to other developers it would be easier to get help 
in some points

b) with git branches you can hack freely into the code without fear of 
breaking anything

I'm personally doing lots of experiments with the Cgl cut generators right 
now, building a dataset of cases where invalid cuts are generated. 
Basically all Cgl cut generators (except the combinatorial ones, like 
clique) fail in some cases, even in well behaved instances. I'm trying to 
build this dataset to apply machine learning techniques to try to predict 
when a generated cut is invalid, which could alleviate the problem. If your 
cut generate could be integrated into the Cgl development branch I would 
be glad to include it in the tests.

Cheers,

Haroldo



-- 
=============================================================
Haroldo Gambini Santos
Department of Computer Science
Universidade Federal de Ouro Preto - UFOP
email: haroldo@ufop.edu.br
home/research page: www.decom.ufop.br/haroldo

**Reply by gnannicini** [reply]
There is a related discussion here: https://link.springer.com/article/10.1007/s12532-013-0057-4
The CglGMI is supposed to be "relatively safe", but no generator is completely safe (except, as you mention, fully combinatorial ones).

**Reply by akazachk** [reply]
Hi @h-g-s, I think my generator is not in a stable enough state for you to test, but I will try to hurry. I have been procrastinating a bit of (tedious) debugging of some hard-to-reproduce behavior in Cbc ––– I must be selecting a weird setting somewhere, as I have an instance for which, in my code, a branch-and-bound node is declared infeasible, but it should be feasible, and this error only happens in my code, not when I try to produce a minimal example.

@gnannicini: I did do some safety tests on a (much) earlier version of my generator, by adapting the `test_cutgen_safety` code for your paper with François and Gérard, and counting "type 1" (cuts that remove an integer-feasible solution) and "type 2" (cuts that yield a primal-infeasible instance) errors from that paper. I wonder if that code still works (it relied on [QSopt_ex](http://www.math.uwaterloo.ca/~bico/qsopt/ex/), if I recall correctly, for which there is a [fork](https://github.com/jonls/qsopt-ex) with some improvements).

**Reply by akazachk** [reply]
As an update, I have added the bug I am working on as issue #338. As @jjhforrest points out, numerical issues are likely playing a role with this instance, but the perplexing thing is that I cannot reproduce the solution process with a small example code that seemingly uses the same settings as my cut generation code.

**Comment by jameshowey** [comment]
I am Jim Howey. I am a back end developer, not an lp expert,  hoping to bootstrap my efforts with some sweet coin-or optimization.
